api/services/CasebookService.js

```diff
@@ -1,52 +1,93 @@
+        };
+        const casebookRequestBase =
+            CasebookService._createBaseRequest(queryParamsObj);
 
-    post(options) {
-        return CasebookService._returnRequestMethods('post', options);
+        return casebookRequestBase.get(customURLs.applicationStatusAPIURL, {
+            params: queryParamsObj,
+        });
     },
 
-    _returnRequestMethods(method, options) {
-        const requestType = options.promise ? requestPromise : request;
-        delete options.promise;
-
-        const authParams = CasebookService._createAuthParams(options.qs);
-        const optionsWithAuthParams = { ...options, ...authParams };
+    getApostilleDownload(apostilleReference) {
+        const queryParamsObj = {
+            timestamp: Date.now().toString(),
+            apostilleReference,
+        };
+        const casebookRequestBase =
+            CasebookService._createBaseRequest(queryParamsObj);
 
-        return requestType[method](optionsWithAuthParams);
+        return casebookRequestBase.get(customURLs.apostilleDownloadAPIURL, {
+            params: queryParamsObj,
+            responseType: 'stream',
+        });
     },
 
-    _createAuthParams(queryParamsObj) {
-        const {
-            hmacKey,
-            casebookCertificate: cert,
-            casebookKey: key,
-        } = sails.config;
+    _createBaseRequest(paramsObj) {
+        const casebookRequestBase = axios.create({
+            baseURL: customURLs.casebookBaseUrl,
+            headers: {
+                'Content-Type': 'application/json; charset=utf-8',
+                'api-version': '4',
+                hash: CasebookService._addHashToHeader(paramsObj),
+            },
+            httpsAgent: new https.Agent({
+                cert,
+                key,
+                keepAlive: true,
+            }),
+            paramsSerializer: queryParamObjToStr,
+            transformRequest: [CasebookService._addHmacToQueryParam],
+        });
+
+        casebookRequestBase.interceptors.response.use(
+            (response) => {
+                if (response.status !== 200) {
+                    throw new Error(`Casebook returned ${response.statusCode}`);
+                }
+                return response;
+            },
+            (error) => {
+                return Promise.reject(error);
+            }
+        );
 
-        const queryStr = apiQueryString.stringify(queryParamsObj);
+        return casebookRequestBase;
+    },
 
+    _addHashToHeader(paramsObj) {
+        const queryStr = queryParamObjToStr(paramsObj);
         const hash = crypto
             .createHmac('sha512', hmacKey)
             .update(Buffer.from(queryStr, 'utf-8'))
             .digest('hex')
             .toUpperCase();
 
-        return {
-            agentOptions: {
-                cert,
-                key,
-            },
-            headers: {
-                hash,
-            },
+        return hash;
+    },
+
+    _addHmacToQueryParam(data, _headers) {
```

Comment:
you don't have to change this now because it would affect the consistency of how things are done in the rest of the app, but you don't have to use `_` or `private` fields when you're writing modules this way. imo it only really matters for (class) OOP 

`module.exports = CasebookService` <- so you're only exposing everything within the casebook object and anything you don't want to expose, you can just take it out of the object. 

The alternative is (although you probably don't have the right transpilation/node ver for this), to export the functions you want to expose. 

```.js
const baseRequest = axios(..) // not exported, private 
export async function getApplicationReference { } // exported, public
```
then, for anything that needs casebook service
```.js
const { getApplicationReference } = require('../CasebookService')
// or import the whole thing if you use multiple methods
```

Reply:
Ah, this is definitely currently possible. I think you can do something like:
```js
module.exports.getApplicationStatus = CasebookService.getApplicationStatus;
```
or
```js
module.exports= {
  getApplicationStatus: CasebookService.getApplicationStatus
};
```

I'll try it out.

Reply:
if you decide to do `module.exports`, there's no point in having the CasebookService object, because you're already selectively picking which methods you're exposing 

```
module.exports = {
  getApplicationStatus
 // not getApplicationStatus: CasebookService.getApplicationService
};
```